app/code/Magento/CatalogGraphQl/Model/Resolver/Products/Query/ResolveCategoryAggregation.php

```diff
@@ -0,0 +1,167 @@
+     *
+     * @param array $categoryFilter
+     * @param Bucket[] $bucketList
+     *
+     * @return Aggregation
+     */
+    public function getResolvedCategoryAggregation(array $categoryFilter, array $bucketList): Aggregation
+    {
+        $categoryBucket = $bucketList[self::CATEGORY_BUCKET] ?? [];
+        $values = $categoryBucket->getValues();
+        $condition = array_key_first($categoryFilter);
+        $searchableCategoryValue = $categoryFilter[$condition] ?? 0;
+
+        if (!$searchableCategoryValue || empty($values)) {
+            return $this->aggregationFactory->create([self::BUCKETS => $bucketList]);
+        }
+
+        $resolvedBucketList = $bucketList;
+
+        try {
+            if (!is_array($searchableCategoryValue)) {
+                $resolvedValues = $this->getValidCategories((int) $searchableCategoryValue, $values);
+            } else {
+                $categoryResolvedValues = [];
+
+                foreach ($searchableCategoryValue as $categoryId) {
+                    $categoryResolvedValues[] = $this->getValidCategories((int) $categoryId, $values);
+                }
+
+                $resolvedValues = call_user_func_array('array_merge', $categoryResolvedValues);;
+            }
+        } catch (NoSuchEntityException $e) {
+            return $this->aggregationFactory->create([self::BUCKETS => $bucketList]);
+        }
+
+        $resolvedCategoryBucket = $this->bucketFactory->create(
+            [
+                'name' => self::CATEGORY_BUCKET,
+                'values' => $resolvedValues
+            ]
+        );
+
+        $resolvedBucketList[self::CATEGORY_BUCKET] = $resolvedCategoryBucket;
+
+        return $this->aggregationFactory->create([self::BUCKETS => $resolvedBucketList]);
+    }
+
+    /**
+     * Check is valid searchable category children.
+     *
+     * @param int $searchableCategoryId
+     * @param AggregationValueInterface[] $aggregationValues
+     *
+     * @return AggregationValueInterface[]
+     *
+     * @throws NoSuchEntityException
+     */
+    private function getValidCategories(int $searchableCategoryId, array $aggregationValues): array
+    {
+        $stareId = (int) $this->storeManager->getStore()->getId();
+        $searchableCategory = $this->categoryRepository->get($searchableCategoryId, $stareId);
+        $childrenList = $searchableCategory->getChildrenCategories();
+        $resolvedList = [];
+        $validChildIdList = [];
+
+        foreach ($childrenList as $child) {
+            if (!$child->getIsActive()) {
+                continue;
+            }
+
+            $validChildIdList[] = $child->getId();
+        }
+
+        foreach ($aggregationValues as $bucketValue) {
+            $childCategoryId = (int) $bucketValue->getValue();
+
+            if (!in_array($childCategoryId, $validChildIdList) ||
+                in_array($childCategoryId, $this->resolvedChildrenIds)
+            ) {
+                continue;
+            }
```

Comment:
Hello @korovitskyi 
Could you please fix it according to PSR-12 requirements and examples https://www.php-fig.org/psr/psr-12/ (see p.5.1)

```suggestion
            if (
                !in_array($childCategoryId, $validChildIdList)
                ||. in_array($childCategoryId, $this->resolvedChildrenIds)
            ) {
                continue;
            }
```